theano/tensor/nnet/nnet.py

```diff
@@ -1485,13 +1383,18 @@ def local_argmax_pushdown(node):
         if x.owner and x.owner.op in (softmax_op, softplus, tensor.exp,
                                       tensor.log, tensor.tanh, sigmoid):
             pre_x, = x.owner.inputs
-            return tensor._max_and_argmax(pre_x, axis)
+            ret = tensor._max_and_argmax(pre_x, axis)
+            copy_stack_trace(pre_x, ret)
```

Comment:
So actually, I think here we should copy the stack trace of `x_max`.
We are replacing `argmax(increasing_op(pre_x))` by `argmax(pre_x)`, so I think what we want to keep track of is the creation of the original `argmax` (not of `pre_x`).